server/src/main/java/org/apache/iotdb/db/qp/Planner.java

```diff
@@ -139,6 +140,40 @@ public PhysicalPlan rawDataQueryReqToPhysicalPlan(
     return physicalGenerator.transformToPhysicalPlan(op, rawDataQueryReq.fetchSize);
   }
 
+  /** convert last data query to physical plan directly */
+  public PhysicalPlan lastDataQueryReqToPhysicalPlan(TSLastDataQueryReq req, ZoneId zoneId)
+      throws QueryProcessException, IllegalPathException {
+    List<String> paths = req.getPaths();
+    long time = req.getTime();
+
+    SelectOperator selectOp = new SelectOperator(SQLConstant.TOK_SELECT, zoneId);
+    FromOperator fromOp = new FromOperator(SQLConstant.TOK_FROM);
+    QueryOperator queryOp = new QueryOperator(SQLConstant.TOK_QUERY);
+
+    selectOp.setLastQuery();
+    selectOp.addSelectPath(new PartialPath(""));
```

Comment:
The data structure of select opeartor was changed, since a PR modified it was merged before. You can see `generate(TSRawDataQueryReq rawDataQueryReq, ZoneId zoneId)` for reference.

```suggestion
    selectOp.setLastQuery();
    selectOp.addResultColumn(new ResultColumn(new TimeSeriesOperand(new PartialPath(""))));
```